frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/SmartScalarSettingsWidgets.tsx

```diff
@@ -35,67 +55,88 @@ export function SmartScalarComparisonWidget({
   const isDisabled = options.length === 1 && !isEmpty(selectedOption);
 
   return (
-    <Menu opened={open} onChange={setOpen} position="bottom-start" shadow="sm">
+    <Menu
+      opened={open}
+      onChange={setOpen}
+      onClose={handleClose}
+      position="bottom-start"
+      shadow="sm"
+      closeOnItemClick={false}
+    >
       <Menu.Target>
-        <ButtonStyled
+        <Button
           data-testid={"comparisons-widget-button"}
+          styles={{ inner: { justifyContent: "space-between" } }}
```

Comment:
Starting point for https://github.com/metabase/metabase/pull/36666#discussion_r1432638800. There does not seem to be an easy to way to get the correct styles using `rightIcon` that will also be compatible with Mantine v7.

WDYT we should do to get the correct styles here but still use `rightIcon`. Note: `getStylesRef` was not working everywhere and does not seem to be a supported way of adjusting styles for component

Reply:
What kind of problem did you have with `getStylesRef`? I think it worked for me, but the tricky part was to use `.` in front of `getStylesRef()` call to get a valid CSS selector:

```tsx
const labelClassName = getStylesRef("label");

const MyStyledComponent = styled(MantineComponent)`
  ${labelClassName} {
    // doesn't work
  }

  .${labelClassName} {
    // works
  }
`
```